### ISSUE TITLE
runtime error of reading health from memory

### ISSUE THREAD

**Issue by Michael-YongWang** [CWHer/Another-RL-Project#2]
Hi there,

The version is 1.06 but I met this error:

```bash
DEBUG:pymem:Process 3296 is being debugged
CRITICAL:root:health_read_addr scan failed
Traceback (most recent call last):
  File "D:\mycode-win\RL-Sekiro\run.py", line 16, in <module>
    env = SekiroEnv()
          ^^^^^^^^^^^
  File "D:\mycode-win\RL-Sekiro\env\sekiro_env.py", line 26, in __init__
    self.memory = Memory()
                  ^^^^^^^^
  File "D:\mycode-win\RL-Sekiro\env\memory.py", line 114, in __init__
    raise RuntimeError()
RuntimeError
```

please help!

Thank you

**Comment by cwher** [comment]
TL;DR: As far as I am concerned, it is unfortunate that this `RuntimeError` may be difficult to solve in your case.

- Root Cause
  `memory.py` searches for a specific memory layout (e.g., the memory should contain a block like `AA BB CC DD`), and this `RuntimeError` indicates that this search attempt has failed.

- Reasons

  1.  `(Likely)` Mismatch of code and game version. Since different versions of the game are likely to have different memory layouts, this code probably can not work on the game version other than `v1.06`. However, in your case, this reason seems to be ruled out.

  2.  `(Unlikely)` Even in the same game version, the memory layout may be different on different machines. I am not sure whether this would happen, since this code once worked well on 3 Windows PC. To solve this issue, one may need to use `Cheat Engine` and do reverse engineering to find the memory layout on a specific machine. It is unfortunate that all of the authors currently do not have this game installed on our machines, and even if we have, we may find that the memory layout is different from yours.
